hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFileSystem.java

```diff
@@ -1926,12 +1926,41 @@ enum RenameStrategy {
     SAME_FILESYSTEM_ACROSS_MOUNTPOINT
   }
 
+  private void closeChildFileSystems(FileSystem fs) throws IOException {
+    if (fs != null) {
+      FileSystem[] childFs = fs.getChildFileSystems();
+      for (FileSystem child : childFs) {
+        if (child != null) {
+          String disableCacheName = String.format("fs.%s.impl.disable.cache",
+              child.getUri().getScheme());
+          if (config.getBoolean(disableCacheName, false)) {
+            child.close();
```

Comment:
do failures need to be handled here? if so: what?

any fs set to delete files on close may raise exceptions here from permissions...should they be ignored, fail fast (as this pr does) or should we try best effort to close the others.

The simple patch here is better than today, so i'm not too worried -and that delete on close stuff isn't something to use in production

Reply:
Thanks for your advice. I will catch IOException here just like `ViewFileSystem.InnerCache#closeAll`.